sdk-wallet/user_wallet_provider.md

```diff
@@ -0,0 +1,37 @@
+## UserWalletProvider
+
+```
+class UserWalletProvider:
+    // Should not throw.
+    generate_keypair(): (ISecretKey, IPublicKey)
+
+    // Can throw:
+    // - ErrInvalidSecretKey
+    sign(data: bytes, secret_key: ISecretKey): bytes
+
+    // Can throw:
+    // - ErrInvalidPublicKey
+    // - ErrInvalidSignature
+    verify(data: bytes, signature: bytes, public_key: IPublicKey): bool
+
+    // Can throw:
+    // - ErrInvalidSecretKey
+    create_secret_key_from_bytes(data: bytes): ISecretKey
+
+    // Can throw:
+    // - ErrInvalidPublicKey
+    create_public_key_from_bytes(data: bytes): IPublicKey
+
+    // Can throw:
+    // - ErrInvalidSecretKey
+    compute_public_key_from_secret_key(secret_key: ISecretKey): IPublicKey
+
+    // Should not throw.
+    generate_mnemonic(): Mnemonic
+
+    // Can throw:
+    // - ErrInvalidMnemonic
+    derive_secret_key_from_mnemonic(mnemonic: Mnemonic, address_index: number = 0, passphrase: string = ""): ISecretKey
```

Comment:
I see this as part of the mnemonic definition. The possibility to convert itself to some other part. Do you think it is worth moving there and keeping only the `generate_mnemonic` function here?
we have L5 on mnemonic.md :D

Reply:
Indeed, moved.